cpp/src/cylon/table.hpp

```diff
@@ -407,6 +407,20 @@ Status Equals(const std::shared_ptr<cylon::Table>& a, const std::shared_ptr<cylo
 
 Status DistributedEquals(const std::shared_ptr<cylon::Table> &a, const std::shared_ptr<cylon::Table> &b, bool& result, bool ordered = true);
 
+Status Repartition(const std::shared_ptr<cylon::Table>& table, 
+                   const std::vector<int64_t>& rows_per_partition,
+                   const std::vector<int>& receive_build_rank_order,
+                   std::shared_ptr<Table> *output);
+
+// TODO: change this to reference
+
+Status Repartition(const std::shared_ptr<cylon::Table>& table,
+                   const std::vector<int64_t>& rows_per_partition,
+                   std::shared_ptr<cylon::Table> *output);
+
+Status Repartition(const std::shared_ptr<cylon::Table>& table,
```

Comment:
we may remove this function. Previous function can get rows_per_partition with default value of empty vector. If the vector is empty, it can perform even repartitioning.